src/test/java/edu/kis/powp/jobs2d/UsageMonitorTest.java

```diff
@@ -0,0 +1,74 @@
+package edu.kis.powp.jobs2d;
+
+import edu.kis.powp.jobs2d.command.*;
+import edu.kis.powp.jobs2d.features.UsageMonitor;
+
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class UsageMonitorTest implements ActionListener {
+
+    private Job2dDriver driver;
+
+    public UsageMonitorTest(Job2dDriver driver)
+    {
+        this.driver = driver;
+    }
+
+    public UsageMonitorTest() {
+
+    }
+
+    private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    @Override
+    public void actionPerformed(ActionEvent actionEvent) {
+        logger.info("Testing Usage Monitor");
+        UsageMonitor um = new UsageMonitor(this.driver);
+
+        List<DriverCommand> driverCommands = new ArrayList<>();
+        driverCommands.add(new SetPositionCommand(-20, -50));
+        driverCommands.add(new OperateToCommand(-20, -50));
+        driverCommands.add(new SetPositionCommand(-20, -40));
+        driverCommands.add(new OperateToCommand(-20, 50));
+        driverCommands.add(new SetPositionCommand(0, -50));
+        driverCommands.add(new SetPositionCommand(-20, -50));
+        driverCommands.add(new OperateToCommand(-20, -50));
+        driverCommands.add(new SetPositionCommand(-20, -40));
+        driverCommands.add(new OperateToCommand(-20, 50));
+        driverCommands.add(new SetPositionCommand(0, -50));
+        driverCommands.add(new OperateToCommand(0, -50));
+        driverCommands.add(new SetPositionCommand(0, -40));
+        driverCommands.add(new OperateToCommand(0, 50));
```

Comment:
You could extract it to separate commands factory, add load option (setCurrentCommand) to GUI and use getCurrentCommand to apply visitor to any command.